### PR TITLE
 Issue  #1979: Add generation of JSON for widget specs which is then used to create MD.

### PR DESCRIPTION
**Pull request by reggied**

### CODE REVIEW

**Comment by jasongrout** [comment]
I extended the program a bit to have an option to generate a json file, and also checked in the current json file. This is so people can read in that json file and do the programmatic checking they want.

@SylvainCorlay - could you look at this, and in particular look at the JSON spec generated, and let us know if it looks good to you?

File: packages/schema/jupyterwidgetmodels.latest.md [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-This is a description of the model state for each widget in the core Jupyter widgets library. The model ID of a widget is the id of the comm object the widget is using. A reference to a widget is serialized to JSON as a string of the form `"IPY_MODEL_<MODEL_ID>"`, where `<MODEL_ID>` is the model ID of a previously created widget of the specified type.
+This is a description of the model state for each widget in the core Jupyter
+widgets library. The model ID of a widget is the id of the comm object the
+widget is using. A reference to a widget is serialized to JSON as a string of
+the form `"IPY_MODEL_<MODEL_ID>"`, where `<MODEL_ID>` is the model ID of a
+previously created widget of the specified type.
 
-This model specification is for ipywidgets 7.4.*, @jupyter-widgets/base 1.1.*, and @jupyter-widgets/controls 1.4.*.
+This model specification is for ipywidgets 7.4.*, @jupyter-widgets/base 1.1.*,
+and @jupyter-widgets/controls 1.4.*.
```
Comment (SylvainCorlay) [inline packages/schema/jupyterwidgetmodels.latest.md]: This probably will require a rebase.

**Comment by SylvainCorlay** [comment]
The JSON latest spec probably needs to be regenerated.

**Comment by SylvainCorlay** [comment]
PS: @reggied, FYI you did not use the same email for your commits as the one that is registered for your GitHub account.

In order to be creditted by GitHub for these commits, either we can rebase an use the same email as in https://github.com/django-salesforce/django-salesforce/commit/0d4a4de143853b6326f78796d74e35800e57196b.patch or add this email to the GitHub account.

**Comment by SylvainCorlay** [comment]
Rebased.

**Comment by SylvainCorlay** [comment]
Thanks @reggied!

Sorry for taking so long. I am merging the PR.

**Pull request merged**